### ISSUE TITLE
Connecting Plugin to Subplots

### ISSUE THREAD

**Issue by avipersin** [jakevdp/mpld3#322]
Is it possible to connect a plugin to all subplots in a figure?  When I try to connect to a figure with more than one subplot, it only applies the plugin to the first subplot.

**Comment by aflaxman** [comment]
It should be possible.  Can you create a minimal example of what you have tried that shows how it goes wrong? StackOverflow has some good advice on creating minimal examples: http://stackoverflow.com/help/mcve

**Comment by avipersin** [comment]
If you use this test script: https://gist.github.com/avipersin/7f27be119f9a602b4e54

with this date formatting plugin:  https://gist.github.com/avipersin/6127f5e655c8e06fbc5a

You will see the plugin is only being applied to the first of two subplots.  Tested using most recent version of mpld3 forked from the repo (0.3) and python 3.4.

**Comment by aflaxman** [comment]
I think you should use another `PointHTMLTooltip` instance to put tooltips on the second panel.  For your `DateFormatter` plugin, you could loop over the axes to make them all dates, but I suggest you instead pass in an axis and just set that one to use dates, so that you can also support plots where only some subplots have date formatting.

**Comment by avipersin** [comment]
Can you provide a minimal working example?

When I try to loop over the axes and add the plugin to each one, I'm getting  "ValueError: plugins.connect: first argument must be a figure".  It seems the connect function expects a figure, not an Axes instance.

**Comment by jakevdp** [comment]
Yes, the first argument must be a figure. As currently implemented, mpld3 doesn't really handle multiple versions of the same plugin on the same figure – as you saw, it simply overwrites them.

The best way forward right now would be to write a single plugin that can handle multiple axes – probably not what you wanted to hear.

**Comment by avipersin** [comment]
Thanks @jakevdp for your response, I have taken care of it on the JavaScript plugin end by iterating over the axes.
